File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
  let sum = 0;
  let num = x;

  while (num != 0) {
    sum += num % 10;
    num = Math.floor(num / 10);
  }

  if (x % sum === 0) {
    return true;
  } else {
    return false;
  }
}

File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        int sum = 0;
        String str = Integer.toString(x);
        for(int i=0; i<str.length(); i++) {
            sum += Integer.parseInt(String.valueOf(str.charAt(i)));
        }
        if(x%sum == 0) {
                answer = true;
        }else {
                answer = false;
        }
        return answer;
    }
}

